src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/Consumers/SingleKafkaConsumer.cs

```diff
@@ -0,0 +1,49 @@
+namespace MassTransit.KafkaIntegration.Consumers
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Confluent.Kafka;
+
+
+    public class SingleKafkaConsumer<TKey, TValue> :
+        IKafkaConsumer
+    {
+        readonly IConsumer<TKey, TValue> _consumer;
+        readonly KafkaHandler<TKey, TValue> _handler;
+        readonly Task _task;
+
+        public SingleKafkaConsumer(IConsumer<TKey, TValue> consumer, KafkaHandler<TKey, TValue> handler, CancellationToken cancellationToken)
+        {
+            _consumer = consumer;
+            _handler = handler;
+            _task = Task.Run(() => Consume(cancellationToken));
```

Comment:
I don't like how this is started. I think each consumer should be an `Agent`, and those agents should be added to a supervisor. In fact, the AddConsumeAgent() is probably a good one that's on the connection context of other transports.